### ISSUE TITLE
config:store:get does not read config.php or env.php

### ISSUE THREAD

**Issue by convenient** [netz98/n98-magerun2#712]
## Current Implementation

When you have config stored in `env.php` and `config.php` then using `config:store:get` can produce values which arent actually used on the site.

For example putting into `env.php`

```php
    'system' => [
        'default' => [
            'admin' => [
                'security' => [
                    'session_lifetime' => 31536000
                ]
            ],
        ]
    ]
];
```

Then running `config:store:get` we get the value from `core_config_data` which won't actually be used in the system.
```
$ magerun2 config:store:get admin/security/session_lifetime
+---------------------------------+---------+----------+-------+
| Path                            | Scope   | Scope-ID | Value |
+---------------------------------+---------+----------+-------+
| admin/security/session_lifetime | default | 0        | 86400 |
+---------------------------------+---------+----------+-------+
```

However magentos config:show command properly reads it
```
$ php bin/magento config:show admin/security/session_lifetime
31536000
```

If you remove the value from `env.php` then magentos `config:show` properly falls back to `core_config_data`

```
php bin/magento config:show admin/security/session_lifetime
86400
```

## Suggested Enhancement
It would be neat if we could have `config:store:get` read `default.xml`, `config.php`, and `env.php` appropriately.

I really like mageruns ability to search with wildcards like `'%yotpo%'` but we cant trust the output of that based on the hardcoded values. 

I'm working around this by  `php bin/magento config:show | grep yotpo`

## Expected Benefits

More truthful values of the actual configuration used on the site. I'm not sure how easy or difficult this is, or whether it would end up as a copy of `config:show` but with differing pattern matching after the full list of config has been worked out?

**Comment by cmuench** [comment]
@convenient Yes, that would be a interesting enhancement. Maybe it's a bit difficult. Currently we use the DB to query the table.
In the new scenario we need to search in the merged config data.

**Comment by convenient** [comment]
@cmuench yes thats what I was thinking which makes the sql syntax a bit difficult. For now i will use magento core `config:show` and try to remember a little `grep -E` regex to replace `%` wildcards :)
